src/omnisharpDownload.ts

```diff
@@ -0,0 +1,120 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+ 
+'use strict';
+
+import * as fs from 'fs-extra-promise';
+import * as path from 'path';
+import * as tmp from 'tmp';
+
+const Decompress = require('decompress');
+const Github = require('github-releases');
+
+const OmnisharpRepo = 'OmniSharp/omnisharp-roslyn';
+const OmnisharpVersion = 'v1.9-alpha3';
+const DefaultInstallLocation = path.join(__dirname, '../.omnisharp');
+
+tmp.setGracefulCleanup();
+
+function getOmnisharpAssetName(): string {
+    switch (process.platform) {
+        case 'win32':
+            return 'omnisharp-win-x64-dnx451.zip';
```

Comment:
Yeah our worry, because we've been bitten by it before with dnx, is that without a standalone runtime to boot on top of, we have to do more work from an editor perspective to get things up and running.

* Check for the CLI
* What locations are "standard"
* Is there an environment variable to get those locations? (dnx had one)
* What cli versions are installed?
* What cli version do we need?
* Do we trust new cli's?
  * Minor versions only?

If we can't find the cli then we have to download that, then download the OmniSharp package.  It's just a whole ton more work, than just running `OmniSharp.exe`.  Hence why we package them as standalone copies.

As I noted on our slack, I'm not adverse to having a `portable` version kicking around, and the plan is to add a new portable drop, but for anyway now we're going to keep the standalone copies.